### PR TITLE
Displaying articles on home page and profile pages

### PR DESCRIPTION
**Pull request by AmberWilkie**
Allow logged-in and non-logged-in users to see articles of all users.
Allow for users to view each others' profiles and non-logged-in users to view profiles.

PT: https://www.pivotaltracker.com/story/show/130181839

Ok, quiz time!

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def create
     if user_signed_in?
       @user = User.find(params[:user_id])
       @article = @user.articles.create(article_params)
-      redirect_to user_article_path(@user.id, @article.id)
+      redirect_to user_articles_path(@user.id, @article.id)
```
Comment (diraulo) [inline app/controllers/articles_controller.rb]: using just `@user` or `@article` should be enough in principle, the `id` will be extracted by the helper method and used to generate the link

File: app/views/articles/index.html.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <h1 class="content-head is-center"> Articles </h1>
 <% current_user.articles.each do |article| %>
   <ul>
-    <li><%= link_to article.title, user_article_path(current_user.id, article.id) %></li>
+    <li><%= link_to article.title, user_articles_path(current_user.id, article.id) %></li>
```
Comment (diraulo) [inline app/views/articles/index.html.erb]: same thing here. And also you're generating the link to a single article so you might want to use `user_article_path` rather

File: features/homepage.feature [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+    | Two     | Two awesome text |
+    | Three   | Three good texts |
+
 
 Scenario: Seeing front-page bits
-  Given I am logged in as "anna@random.com"
+  Given I am logged in as "hanna@random.com"
   When I visit the "home" page
   Then I should see "Logout"
+  And I should see "Current Articles"
```
Comment (diraulo) [inline features/homepage.feature]: You could dry this by using a single step that takes a table of all the content you want to see on the page. i.e

``` gherkin
Then I should see:
  | content |
  | item 1  |
  | item 2  |
  | etc..   |
```

File: features/homepage.feature [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+  And I should see "One"
+  And I should see "Two"
+  And I should see "Three"
+
+Scenario: viewing articles when not logged in
+  Given I am not logged in
+  And I am on the "home" page
+  And I should see "Current Articles"
```
Comment (diraulo) [inline features/homepage.feature]: same here

File: spec/factories/users.rb [hunk 5]
```diff
@@ -1,8 +1,10 @@
 FactoryGirl.define do
+    @random = rand(1..100)
   factory :user do
     name "My Name"
-    email "My@Email.com"
+    email "My@Email#{@random}.com"
```
Comment (diraulo) [inline spec/factories/users.rb]: in future consider using [ffaker gem](https://github.com/ffaker/ffaker) to generate random email address of any other fake information for testing.

**Review by diraulo — CHANGES_REQUESTED** [review]

File: spec/factories/users.rb (hunk 5 above)
Comment (AmberWilkie) [reply spec/factories/users.rb]: I will indeed do this...

**Pull request merged**